packages/vite/src/node/plugins/html.ts

```diff
@@ -523,6 +543,14 @@ export async function applyHtmlTransforms(
   return html
 }
 
+const importRE = /\bimport\s*("[^"]*[^\\]"|'[^']*[^\\]');*/g
+const commentRE = /\/\*[\s\S]*?\*\/|\/\/.*$/gm
+function isEntirelyImport(code: string) {
+  // only consider "side-effect" imports, which match <script type=module> semantics exactly
+  // the regexes will remove too little in some exotic cases, but false-negatives are alright
+  return !code.replace(importRE, '').replace(commentRE, '').trim().length
```

Comment:
Couple of potential issues here that may or may not come up in practice:

1. If the code included comments then that would fail this check, even though the modules _could_ be inlined.
2. Since this is relying on a regexp rather than an actual parser it could get tripped up with code like:

```js
const someString = 'import "x"'
```

Wonder if it makes sense to use `es-module-lexer` as is used elsewhere in the codebase to fix Issue 2?

Reply:
This was also an issue that was raised when we discussed the implementation.

About 2, that shouldn't be an issue becasue the resulting length after removing the content of the string will be larger than zero (except if you have discovered an issue in the regex that needs to be fixed)

About 1, the regex is applied to code generated by rollup (so it is somewhat controlled), and the minifier has already been applied so I don't think we would get false negatives here. 

I think that we should merge this PR, but be open to improve it in the future. If you find later a failing test case, we could add that and migrate to a better implementation for that util. A lexer could be used if we have some heuristic about only applying it if the file is smaller than a threshold since it only applies when there are a bunch of imports (so we avoid paying for it for large modules). What do you think?

Reply:
Yeup that sounds good to me. Was most concerned about #2 but I think you are right that it should not be an issue.